src/crimson/admin/osd_admin.cc

```diff
@@ -131,4 +132,34 @@ class AssertAlwaysHook : public AdminSocketHook {
 };
 template std::unique_ptr<AdminSocketHook> make_asok_hook<AssertAlwaysHook>();
 
+/**
+* A Seastar admin hook: fetching the values of configured metrics
+*/
+class SeastarMetricsHook : public AdminSocketHook {
+public:
+ SeastarMetricsHook()  :
+   AdminSocketHook("perf dump_seastar",
+      "",
+      "dump current configured seastar metrics and their values")
+ {}
+ seastar::future<tell_result_t> call(const cmdmap_t& cmdmap,
+             std::string_view format,
+             ceph::bufferlist&& input) const final
+ {
+   std::unique_ptr<Formatter> f{Formatter::create(format, "json-pretty", "json-pretty")};
+   f->open_object_section("perf_dump_seastar");
+   for (const auto& mf : seastar::metrics::impl::get_value_map()) {
+     for (const auto& m : mf.second) {
+       if (m.second && m.second->is_enabled()) {
+         auto metric_function = m.second->get_function();
+         f->dump_float((m.second->get_id().full_name()).c_str(), metric_function().d());
```

Comment:
always floats?

Reply:
A metric value can only be double, int or unsigned int. Once you have an instance of type `metric_value` you can get the actual data using one of the three functions: `d()`, `i()`, or `ui()` namely double, int and unsigned int. Using d() combined with dump_float is the most flexible because if the value is int/unsigned, the output will be an integer/unsigned (without floating points)

Reply:
Not sure I understand. WHy not handle each of the three types separately? from what I see in the formatter code, it
seems that if the type of the added element is double, the number of precision digits will be set for double.

Reply:
I don't think there is a reason to handle each of them separately. Take a look at the doc I shared with you a while back: https://docs.google.com/document/d/1dOLGLVIKzNNeuXXv2iCHvSV5-7sxzFEBl9yWUF893Aw/edit
At the end of it you will find an output example in which you can see that double/float is printed as double/float and int/unsigned is printed as int/unsigned without precision digits, which is exactly what we want.